### PR TITLE
`Development`: Cleanup control flow in client code

### PR DESCRIPTION
**Pull request by aplr**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [ ] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The auto migrator did only syntactically migrate, but leaves semantics and readability behind. This PR refactors a small portion of auto-migrated control flow statements.

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **Enhancements**
  - Improved logic for rendering table rows and pagination in the Audits section.
  - Enhanced display formatting for garbage collector metrics.
  - Refined the system notification management interface for better clarity.
  - Updated user management controls to better reflect user activation status.
  - Adjusted assessment header to conditionally display information based on test run status.
  - Streamlined the display of complaints in the list of complaints section.
  - Optimized loading state indicators across various components for a smoother user experience.
  - Enhanced conditional rendering logic in the student exams and assessment dashboard sections.
  - Improved visual representation of tree structures in the code editor.
  - Refined the display of exercise hints and scores for a more intuitive user interface.
  - Updated the course discussion component to handle loading states more effectively.
  - Polished the course registration modal to improve user interaction during loading states.
  - Enhanced consistency checks and loading indicators throughout the application for better performance feedback.

- **Bug Fixes**
  - Fixed issues with conditional rendering that could affect the visibility of certain elements based on user permissions and system states.

### CODE REVIEW

**Review by valentin-boehm — APPROVED** [review]
Changes look good to me

**Review by yassinsws — APPROVED** [review]
Changes look good to me

File: src/main/webapp/app/exam/manage/student-exams/student-exams.component.html [hunk 1]
```diff
@@ -43,8 +39,6 @@ <h2>
                 <fa-icon [icon]="faExclamationTriangle" class="text-warning"></fa-icon>
                 <span jhiTranslate="artemisApp.studentExams.unlockAllRepositories">Unlock all repositories</span>
             </button>
-        }
-        @if (!localVCEnabled && course?.isAtLeastInstructor) {
```
Comment (b-fein) [inline src/main/webapp/app/exam/manage/student-exams/student-exams.component.html:47]: Check for `!localVCEnabled` accidentally removed now?

**Review by b-fein — APPROVED** [review]
One question, where I think the logic might have changed. Code looks good otherwise.

File: src/main/webapp/app/exam/manage/student-exams/student-exams.component.html (hunk 1 above)
Comment (aplr) [reply src/main/webapp/app/exam/manage/student-exams/student-exams.component.html:47]: Hey, the diff is a bit confusing. If you expand the hidden part, you'll see that there's the same `@if` covering multiple elements now

**Review by b-fein — DISMISSED** [review]
Question resolved, just missed the relevant change.

Code looks good.

**Review by kaancayli — DISMISSED** [review]
Changes look good. Code lgtm

**Review by Strohgelaender — APPROVED** [review]
lgtm

**Review by b-fein — DISMISSED** [review]
code re-approve

**Review by yassinsws — DISMISSED** [review]
Code LGTM

**Review by laurenzfb — DISMISSED** [review]
looks good

**Review by kaancayli — DISMISSED** [review]
Reapprove. LGTM

**Review by milljoniaer — APPROVED** [review]
approve merge commit by @Strohgelaender

**Comment by krusche** [comment]
This PR is too open for a long time and will be closed due to inactivity

**Pull request closed without merging**